src/Parser.js

```diff
@@ -199,6 +200,37 @@ Parser.prototype.handleInfixNodes = function (body, mode) {
     }
 };
 
+/**
+ * Turn an array-like data structure, with rows delimited by \\ and
+ * columns delimited by &, into nested lists of groups.  This method
+ * isn't directly called by the parser, but may be called by some
+ * environments.
+ *
+ * @param {string} mode       the mode to assign to newly created groups
+ * @param {ParseResult} body  the overall parse result to be decomposed
+ * @param {Array}             decomposed data in row major mode
+ */
+Parser.prototype.decomposeArray = function(mode, body) {
```

Comment:
That is where I had them first, but it needs the `ParseNode` constructor, which I hadn't made part of `Parser`'s prototype at that time. Since I later on needed `ParseNode` for the `leftright` group, and moving that to the parser felt even more wrong, I can move this decomposition to the environments. I guess it would be cleaner to export `ParseNode` without reference to the instance, preferably by moving it to a separate file.